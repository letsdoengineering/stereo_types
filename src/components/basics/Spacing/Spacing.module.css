.spacing {
    display: flex;
}

.spacing-single {
    padding: 0.5rem 0;
}

.spacing-double {
    padding: 1rem 0;
}
