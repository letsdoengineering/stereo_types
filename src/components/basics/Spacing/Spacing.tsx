import React from 'react'
import * as spacingStyles from './Spacing.module.css'
import getClassNames from '../../../utils/get-class-names'

type SpacingProps = {
  /** level of spacing to apply */
  size?: 'Single' | 'Double'
  /** children always of type ReactNode */
  children?: React.ReactNode
}

/** Spacing: Returns a sizeable 'spacing' element that creates empty space on a page using a div */
const Spacing: React.FC<SpacingProps> = ({ size = 'Single', children }: SpacingProps) => {
  const classNames = getClassNames({
    defaultClasses: [spacingStyles[`spacing${size}`], spacingStyles['spacing']],
  })
  return <div className={classNames}>{children}</div>
}

export default Spacing
