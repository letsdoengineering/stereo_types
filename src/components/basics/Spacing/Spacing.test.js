import renderer from 'react-test-renderer'
import Spacing from './Spacing'
import { render } from '@testing-library/react'
import { axe } from 'jest-axe'
import * as React from 'react'

describe('Spacing', () => {
    it('has not changed the JSX layout of the page from previous snapshot', () => {
        const tree = renderer.create(<Spacing />).toJSON()

        expect(tree).toMatchSnapshot()
    })

    it('renders the page DOM that is accessible', async () => {
        const { container } = render(<Spacing />)
        const results = await axe(container)

        expect(results).toHaveNoViolations()
    })
})
