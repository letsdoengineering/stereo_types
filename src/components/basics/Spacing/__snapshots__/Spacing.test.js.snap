// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spacing has not changed the JSX layout of the page from previous snapshot 1`] = `
<div
  className="spacingSingle spacing"
/>
`;
